playbooks/common/openshift-cluster/master_facts_tasks.yml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Set config_base if unset for hosted roles
+  set_fact:
+    openshift_config_base: "/etc/origin"
+  when: openshift_config_base is not defined
+
+- name: Set openshift_master_config_dir for hosted roles
+  set_fact:
+    openshift_master_config_dir: "{{ openshift_config_base }}/master"
+  when: openshift_master_config_dir is not defined
+
+- name: Set openshift_master_config_file for hosted roles
+  set_fact:
+    openshift_master_config_file: "{{ openshift_master_config_dir }}/master-config.yaml"
+  when: openshift_master_config_file is not defined
+
+- name: slurp openshift_master_config_file
+  slurp:
+    src: "{{ openshift_master_config_file }}"
+  register: omcf_slurp
+
+- name: Set openshift_master_public_api_url for hosted roles
+  set_fact:
+    openshift_master_public_api_url: "{{ omcf_slurp['content'] | b64decode | from_yaml | json_query('masterPublicURL') }}"
```

Comment:
`(omcf_slurp['content'] | b64decode | from_yaml).masterPublicURL` instead to avoid adding the new dep which I can't find in RHEL7?

Reply:
https://github.com/openshift/openshift-ansible/blob/master/roles/etcd_migrate/tasks/add_ttls.yml#L8 is an example of such.